### PR TITLE
lib.nix refactoring

### PR DESCRIPTION
**Pull request by YorikSar**
Several small improvements to make it look better.

### CODE REVIEW

**Comment by YorikSar** [comment]
Looks like adding `-c true` to `nix develop` works! :)

File: lib.nix
Comment (yannham) [inline lib.nix:194]: I think this can, in theory, have a different semantics than the original code if the input is different from `nixpkgs` but defines both `legacyPackages` and `packages`. In that case, the old code would select the one from `packages`, will the new code will select the one from `legacyPackages`. However, in practice, I have no idea if this situation even makes sense: can it happen that another flake than `nixpkgs` defines a `legacyPackages` output? Even in this case, I'm not sure the previous behavior was the right one either. I just wanted to point this out

**Review by yannham — APPROVED** [review]

File: lib.nix
Comment (YorikSar) [reply lib.nix:194]: Any flake can in theory define both `packages` and `legacyPackages`, but in practice I'm yet to see non-`nixpkgs` flake with `legacyPackages`, because the difference between is quite obscure. That said, hardcoding `nixpkgs` seems worse because if such flake is used somewhere, there would be no way of using it here.

**Comment by YorikSar** [comment]
@yannham I've just rebased the PR on top of master with no changes. GitHub said that it needs a rebase to be merged. I have commit rights, but I try to wait for reviews before merging. Maybe I should've landed this one after your approval and a couple of quiet days.

**Comment by thufschmitt** [comment]
@YorikSar I think it's reasonable for you to merge once you got an approval. Let's keep this process light :)

**Comment by yannham** [comment]
Right, I'll try to just comment - and not approve - on PR that I feel less confident to endorse (if there's Nix dark magic involved, for example), so that we can keep the 1 approval is sufficient.

**Pull request merged**
